### ISSUE TITLE
Can uriAnchor.setAnchor ignore missing arguments?

### ISSUE THREAD

**Issue by dylanmac** [mmikowski/urianchor#10]
Is it possible to update some but not all of the key/values via setAnchor?  For example if I have a fragment like so:
`#!page=profile&subpage=one&subsubpage=two`
and I only want to change the the "subpage" value.  I see that if I use setAnchor like so:
```
$.uriAnchor.setAnchor({
	subpage	: 'three'
});
```
The hash is updated to this:
`#!subpage=three`

I realize I could capture the existing key values via makeAnchor and then insert them into setAnchor but that's a lot of argument management which I hope to avoid.

**Comment by mmikowski** [comment]
I just updated to 1.3.3 which adds the example file changeAnchorPart.js.  Please review.

At some point, I intend to provide a native method in uriAnchor, but this should be a good start for you.

Please close this issue if this is satisfactory :)

**Comment by dylanmac** [comment]
Wow!  Is this something you had thought about including or did you just bang it out?  Regardless, is this the proper usage?

'''
changeAnchorPart({
	key : value
});
'''

**Comment by mmikowski** [comment]
Oh I just banged it out :)  Seriously though, its an example from my book, chapter 4-ish.  And yeah, that's how you use it.  Notice you must set the independent variable if you are going to set it's dependent values.  So changeAnchorPart( { key : value, _key : { sub1 : _2 } }).  Notice the dependent keys are replaced "whole", which probably shouldn't be the case.  But hey, it's a start.

Ultimately, this is a stop-gap.  I want to integrate all of the above into uriAnchor for my current project, but I don't want to change it until I get a tracked nodejs test suite in place.

**Comment by dylanmac** [comment]
I must be using changeAnchorPart incorrectly because all of the hash key/values are getting replaced with the one defined in the function. Here's my usage:

changeAnchorPart({
	filter1	: 'filter-val'
 });

Do I use it within $.uriAnchor.setAnchor?

**Comment by mmikowski** [comment]
My guess is you are trying to replace a dependent key.

Conceptually this is actually quite simple.  Please try the following:

1. Read the uri using anchor_map = $.uriAnchor.makeAnchorMap();
2. Change the desired parameter: anchor_map.some_key = value;
3. Set the anchor: $.uriAnchor.set anchor( anchor_map );

I now recall why changeAnchorPart is not in the core plugin: because the application needs to examine changes and then decide if they are valid.  It may, for example, reset the anchor if the requested parameter combination is not supported.  I should add this to the documentation!

Let me know if that works for you!

**Comment by dylanmac** [comment]
Hi Michael,

Sorry for the long delay.  I got sidetracked onto another project.

I am not trying to replace a dependent key.  However, I am updating a value that uses the pipe as a delimiter, e.g. filter1=EQ|COMM|MM.  I see you said that it might reset if it encounters something invalid.

With that possibility in mind I tried changing the delimiter to a plus(+) sign.  No dice.  It still replaces the hash with the single key/value pair.

The three step guide you provided above DOES work, however.  I suppose I could just use that (though I like the simplicity of changeAnchorPart).

But using the three step process above I am running into one last issue: the setAnchor appears to be encoding the value delimiters so that the key/value pair in the URL ends up looking like this: &filter1=EQ%7CCOMM%7CMM.  Is there anyway to prevent this encoding?

Dylan
